// react-mde/lib/js/commands/linkCommand.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var React = require("react");
var MarkdownUtil_1 = require("../util/MarkdownUtil");
var components_1 = require("../components");
exports.linkCommand = {
    buttonContent: React.createElement(components_1.MdeToolbarIcon, { icon: "link" }),
    buttonProps: { "aria-label": "Insert a link" },
    execute: function (getMarkdownState, setMarkdownState) {
        var _a = getMarkdownState(), text = _a.text, selection = _a.selection;
        var newSelection = MarkdownUtil_1.selectWordIfCaretIsInsideOne({ text: text, selection: selection });
        var _b = MarkdownUtil_1.insertText(text, "[", newSelection.start), newText = _b.newText, insertionLength = _b.insertionLength;
        var finalText = MarkdownUtil_1.insertText(newText, "](url)", newSelection.end + insertionLength).newText;
        setMarkdownState({
            text: finalText,
            selection: {
                start: newSelection.start + insertionLength,
                end: newSelection.end + insertionLength,
            },
        });
    },
};
